modules/core/src/v2/coins/algo.ts

```diff
@@ -285,39 +277,109 @@ export class Algo extends BaseCoin {
           'fee',
           'memo',
           'type',
-          'senderAddress',
-          'voteKeyBase64',
-          'voteLastBlock',
-        ],
-        id,
-        outputs,
-        outputAmount,
-        changeAmount: 0,
-        fee,
-        changeOutputs: [],
-        memo,
-        type,
-        senderAddress,
-        voteKeyBase64,
-        voteLastBlock,
-      };
+          'operations',
+        ];
+
+        const explanationResult: AlgoTransactionExplanation = {
+          displayOrder,
+          id: txJson.id,
+          outputAmount: txJson.amount.toString(),
+          changeAmount: '0',
+          outputs,
+          changeOutputs: [],
+          fee: txJson.fee,
+          memo: txJson.note,
+          type: tx.type,
+          operations,
+        };
+
+        if (txJson.tokenId) explanationResult.tokenId = txJson.tokenId;
+
+        return explanationResult;
+      }
+
+      if (tx.type === accountLib.BaseCoin.TransactionType.WalletInitialization) {
+        const displayOrder = [
+          'id',
+          'fee',
+          'memo',
+          'type',
+          'voteKey',
+          'selectionKey',
+          'voteFirst',
+          'voteLast',
+          'voteKeyDilution',
+        ];
+
+        const explanationResult: AlgoTransactionExplanation = {
+          displayOrder,
+          id: txJson.id,
+          outputAmount: '0',
+          changeAmount: '0',
+          outputs: [],
+          changeOutputs: [],
+          fee: txJson.fee,
+          memo: txJson.note,
+          type: tx.type,
+          voteKey: txJson.voteKey,
+          selectionKey: txJson.selectionKey,
+          voteFirst: txJson.voteFirst,
+          voteLast: txJson.voteLast,
+          voteKeyDilution: txJson.voteKeyDilution,
+        };
+        return explanationResult;
+      }
     })
       .call(this)
       .asCallback(callback);
   }
 
+  /**
+   * returns if a tx is an enable or disable token tx
+   * @param tx - tx in JSON format
+   * @returns true if it's a token tx
+   */
+  getTokenTxType(tx: { amount: string; closeRemainderTo: string; from: string; to: string }): string {
```

Comment:
Another good opportunity to use the `TransferType` type suggested in a previous comment. Whenever you have a hard coded string in your source code that appears in multiple places, you should use one of these types instead of declaring as `string`.